### PR TITLE
Remove old scenarios when loading legacy scenarios

### PR DESCRIPTION
**Pull request by johnimalmberg**

### CODE REVIEW

File: BridgeCareApp/VuejsApp/src/services/scenario.service.ts [hunk 1]
```diff
@@ -22,36 +22,76 @@ export default class ScenarioService {
...
+                                if (scenariosToAdd.length != 0) {
+                                    nodejsAxiosInstance.post('api/AddMultipleScenarios', scenariosToAdd)
+                                        .then((res: AxiosResponse<Scenario[]>) => {
+                                            if (hasValue(res)){
+                                                return resolve(res);
+                                            }
+                                        })
+                                        .catch((error: any) => resolve(error.response));
+                                }
+                                this.removeMongoScenarios(scenariosToRemove).then((res: AxiosResponse) => {
```
Comment (ppator1) [inline BridgeCareApp/VuejsApp/src/services/scenario.service.ts]: What happens if there are no scenarios to remove? Might we need a logical check to see if we need to run this step?

File: BridgeCareApp/VuejsApp/src/services/scenario.service.ts [hunk 2]
```diff
@@ -22,36 +22,76 @@ export default class ScenarioService {
...
+        return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
+            axiosInstance.get('api/GetScenarios')
+                .then((responseLegacy: AxiosResponse<Scenario[]>)=>{
+                    if (hasValue(responseLegacy)){
+                    this.getMongoScenarios()
+                        .then((responseMongo: AxiosResponse<Scenario[]>)=>{
+                            if (hasValue(responseMongo)) {
+                                var scenariosToAdd = this.getMissingScenarios(responseLegacy.data, responseMongo.data);
+                                var scenariosToRemove = this.getMissingScenarios(responseMongo.data, responseLegacy.data);
+                                if (scenariosToAdd.length != 0) {
```
Comment (ppator1) [inline BridgeCareApp/VuejsApp/src/services/scenario.service.ts]: This is just a personal preference of mine, but I would use '> 0' as my logical check.

File: BridgeCareApp/VuejsApp/src/services/scenario.service.ts [hunk 3]
```diff
@@ -22,36 +22,76 @@ export default class ScenarioService {
...
+     * Updates mongodb to match the scenarios from the legacy database
+     */
+    static getLegacyScenarios(): AxiosPromise {
+        return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
+            axiosInstance.get('api/GetScenarios')
+                .then((responseLegacy: AxiosResponse<Scenario[]>)=>{
+                    if (hasValue(responseLegacy)){
+                    this.getMongoScenarios()
+                        .then((responseMongo: AxiosResponse<Scenario[]>)=>{
+                            if (hasValue(responseMongo)) {
```
Comment (ppator1) [inline BridgeCareApp/VuejsApp/src/services/scenario.service.ts]: Responses from mongo aren't as cut & dry as responses from the C# application. I assumed you tested, but if you look at other Vue services there is a required conversion from mongo's data structure to the one used by Vue.

Comment (johnimalmberg) [reply BridgeCareApp/VuejsApp/src/services/scenario.service.ts]: That conversion takes place in the caller of this function (the getLegacyScenarios action from scenario.module.ts).

Should I move the conversion into the service, or leave it as part of the action?

File: BridgeCareApp/VuejsApp/src/services/scenario.service.ts (hunk 1 above)
Comment (johnimalmberg) [reply BridgeCareApp/VuejsApp/src/services/scenario.service.ts]: The removeMongoScenarios function does nothing when given an empty array.

Should I add an explicit check around the call for the sake of readability?

Comment (ppator1) [reply BridgeCareApp/VuejsApp/src/services/scenario.service.ts]: If it isn't causing an issues and is already not running if the array is empty then we can leave it as is.

File: BridgeCareApp/VuejsApp/src/services/scenario.service.ts (hunk 3 above)
Comment (ppator1) [reply BridgeCareApp/VuejsApp/src/services/scenario.service.ts]: I didn't notice that function call. This is fine.

**Pull request merged**